EpiAware/src/EpiInfModels/InfectionODEProcess.jl

```diff
@@ -0,0 +1,182 @@
+u0 = [1.0]
+p = [r]
+params = ODEParams(u0 = u0, p = p)
+
+# Define the ODE problem using SciML
+# We use a simple exponential growth model
+
+function expgrowth(du, u, p, t)
+    du[1] = p[1] * u[1]
+end
+prob = ODEProblem(expgrowth, u0, (0.0, 10.0), p)
+
+# Define the InfectionODEProcess
+
+expgrowth_model = InfectionODEProcess(prob::ODEProblem; ts = 0:1:10,
+        solver = Tsit5(),
+        sol2infs = sol -> sol[1, :])
+
+# Generate the latent infections
+I_t = generate_latent_infs(expgrowth_model, params)()
+
+# output
+
+11-element Vector{Float64}:
+ 1.0
+ 1.1040895124087677
+ 1.2190137467993492
+ 1.3459001375697022
+ 1.4859941865014936
+ 1.640671113705054
+ 1.8114471151863056
+ 1.9999990356297939
+ 2.2081789476865237
+ 2.438027196361022
+ 2.6918002758361723
+```
+"""
+struct InfectionODEProcess{P, T, S, F} <: EpiAwareBase.AbstractTuringEpiModel where {
+    P <: ODEProblem,
+    T,
+    S,
+    F <: Function}
+    prob::P
+    ts::Vector{T}
+    solver::S
+    sol2infs::F
+end
+
+function InfectionODEProcess(prob::ODEProblem; ts,
+        solver = AutoTsit5(Rosenbrock23()),
+        sol2infs = sol -> sol[end, :])
+    P = typeof(prob)
+    T = eltype(ts)
+    S = typeof(solver)
+    F = typeof(sol2infs)
+    return InfectionODEProcess{P, T, S, F}(prob, ts, solver, sol2infs)
+end
+
+"""
+Implement the `generate_latent_infs` function for the `InfectionODEProcess` model.
+
+Constructs a `Turing` model to generate latent infections using the specified epidemiological
+model and parameters.
+
+# Arguments
+- `epi_model::InfectionODEProcess`: The `InfectionODEProcess` model containing the problem definition, time steps, solver, and solution-to-infections transformation function.
+- `params::ODEParams`: The initial conditions (`u0`) and parameters (`p`) for the ODE problem.
+
+# Generated quantities
+- `I_t`: The latent infections generated by solving the ODE problem with the specified parameters.
+
+# Details
+This function remakes the ODE problem with the provided initial conditions and parameters, solves it using the specified solver, and then transforms the solution into latent infections using the `sol2infs` function.
+
+# Example usage
+
+```julia
+using EpiAware, OrdinaryDiffEq
+r = log(2) / 7 # Growth rate corresponding to 7 day doubling time
+u0 = [1.0]
+p = [r]
+params = ODEParams(u0 = u0, p = p)
+
+# Define the ODE problem using SciML
+# We use a simple exponential growth model
+
+function expgrowth(du, u, p, t)
+    du[1] = p[1] * u[1]
+end
+prob = ODEProblem(expgrowth, u0, (0.0, 10.0), p)
+
+# Define the InfectionODEProcess
+
+expgrowth_model = InfectionODEProcess(prob::ODEProblem; ts = 0:1:10,
+        solver = Tsit5(),
+        sol2infs = sol -> sol[1, :])
+
+# Generate the latent infections
+I_t = generate_latent_infs(expgrowth_model, params)()
+```
+"""
+@model function EpiAwareBase.generate_latent_infs(
+        epi_model::InfectionODEProcess, params::ODEParams)
```

Comment:
As written this doesn't look like it will work within the generate_epiaware framework (https://github.com/CDCgov/Rt-without-renewal/blob/c5f47ff512df19f919a477e2af0cd2d03413602e/EpiAware/src/EpiAwareUtils/turing-methods.jl#L14)

We have agreed elsewhere to update the infection processes so that they don't depend on a latent model (once they are composable). 

One option could be to encode the initial conditions and parameters as submodel dispatched via a field in the ODEProcess struct (i.e ODEParams is a field of ODEProcess) and have a deterministic default (i.e the current struct).